code/model/DMSDocumentSet.php

```diff
@@ -348,4 +355,71 @@ protected function validate()
         }
         return $result;
     }
+
+    public function canView($member = null)
+    {
+        $extended = $this->extendedCan(__FUNCTION__, $member);
+        if ($extended !== null) {
+            return $extended;
+        }
+        return $this->getDMSDocumentSetPermissions($member);
+    }
+
+    public function canCreate($member = null)
+    {
+        $extended = $this->extendedCan(__FUNCTION__, $member);
+        if ($extended !== null) {
+            return $extended;
+        }
+        return $this->getDMSDocumentSetPermissions($member);
+    }
+
+    public function canEdit($member = null)
+    {
+        $extended = $this->extendedCan(__FUNCTION__, $member);
+        if ($extended !== null) {
+            return $extended;
+        }
+        return $this->getDMSDocumentSetPermissions($member);
+    }
+
+    public function canDelete($member = null)
+    {
+        $extended = $this->extendedCan(__FUNCTION__, $member);
+        if ($extended !== null) {
+            return $extended;
+        }
+        return $this->getDMSDocumentSetPermissions($member);
+    }
+
+    /**
+     * Checks if a then given (or logged in) member is either an ADMIN, SITETREE_EDIT_ALL or has access
+     * to the DMSDocumentAdmin module, in which case permissions is granted.
+     *
+     * To extend use the following from within an Extension subclass:
+     *
+     * <code>
+     * public function updateDMSDocumentSetPermissions($result)
+     * {
+     *     // Do something here
+     * }
+     * </code>
+     * @param Member $member
+     * @return bool
+     */
+    public function getDMSDocumentSetPermissions(Member $member = null)
```

Comment:
Given we're in the context of a DMSDocumentSet I don't see much benefit from adding that into the method naming again - using it would then mean writing `DMSDocumentSet::singleton()->getDMSDocumentSetPermissions()` over something like `DMSDocumentSet::singleton()->getPermissions()`.

Arguably it's also a single permission (yes/no) rather than permissions - maybe this method could be renamed to `getGlobalPermission` or something like that?